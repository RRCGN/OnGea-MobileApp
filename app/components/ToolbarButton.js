/**
 * Button for use in Toolbar.
 * @flow
 */

import React from 'react'
import { View, StyleSheet } from 'react-native'
import TouchableItem from 'react-navigation/src/views/TouchableItem'
import PlatformIcon from './PlatformIcon'


type Props = {
  androidIcon: string,
  iosIcon: string,
  floating?: boolean,
  onPress?: () => void
}

const ToolbarButton = ({
  androidIcon,
  iosIcon,
  floating = false,
  onPress = () => { }
}: Props) => {
  return (
    <TouchableItem
      onPress={onPress}
      delayPressIn={0}
      style={styles.container}
      borderless
    >
      <View style={styles.container}>
        <PlatformIcon
          androidIcon={androidIcon}
          iosIcon={iosIcon}
          size={24}
          color="white"
          style={[styles.icon, floating && styles.floating]}
        />
      </View>
    </TouchableItem>
  )
}

export default ToolbarButton


const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    padding: 6
  },
  floating: {
    textShadowColor: 'rgba(0, 0, 0, 0.4)',
    textShadowRadius: 2,
    textShadowOffset: { width: 0, height: 1 }
  }
})
