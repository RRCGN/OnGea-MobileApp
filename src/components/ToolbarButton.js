/**
 * Button for use in Toolbar.
 * @flow
 */

import React from 'react'
import { View, StyleSheet } from 'react-native'
import TouchableItem from 'react-navigation/src/views/TouchableItem'
import { Colors } from '../utils/constants'
import PlatformIcon from './PlatformIcon'


type Props = {
  androidIcon: string,
  iosIcon: string,
  iconColor: string,
  floating?: boolean,
  onPress?: () => void
}

const ButtonToolbar = ({
  androidIcon,
  iosIcon,
  iconColor,
  floating = false,
  onPress = () => { }
}: Props) => {
  return (
    <TouchableItem
      onPress={onPress}
      delayPressIn={0}
      style={styles.container}
      borderless
    >
      <View style={styles.container}>
        <PlatformIcon
          androidIcon={androidIcon}
          iosIcon={iosIcon}
          size={24}
          color={iconColor ? iconColor : Colors.LIGHT_PRIMARY}
          style={[ styles.icon, floating && styles.floating ]}
        />
      </View>
    </TouchableItem>
  )
}

export default ButtonToolbar


const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    padding: 6
  },
  floating: {
    textShadowColor: 'rgba(0, 0, 0, 0.4)',
    textShadowRadius: 2,
    textShadowOffset: { width: 0, height: 1 }
  }
})
