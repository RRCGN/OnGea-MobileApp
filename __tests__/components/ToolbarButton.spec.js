import { View } from 'react-native'
import React from 'react'
import ToolbarButton from '../../app/components/ToolbarButton'

import renderer from 'react-test-renderer'

jest.mock('react-navigation/src/views/TouchableItem', () => 'TouchableOpacity')

describe('<ToolbarButton />', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <ToolbarButton
        androidIcon="settings"
        iosIcon="ios-cog-outline"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly as floating button', () => {
    const tree = renderer.create(
      <ToolbarButton
        floating={true}
        androidIcon="settings"
        iosIcon="ios-cog-outline"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with onPress', () => {
    const tree = renderer.create(
      <ToolbarButton
        onPress={() => {}}
        androidIcon="settings"
        iosIcon="ios-cog-outline"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
