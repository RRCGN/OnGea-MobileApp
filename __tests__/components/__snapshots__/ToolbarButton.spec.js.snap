exports[`<ToolbarButton /> renders correctly 1`] = `
<TouchableOpacity
  borderless={true}
  delayPressIn={0}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 6,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 6,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 24,
          },
          Array [
            undefined,
            false,
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`<ToolbarButton /> renders correctly as floating button 1`] = `
<TouchableOpacity
  borderless={true}
  delayPressIn={0}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 6,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 6,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 24,
          },
          Array [
            undefined,
            Object {
              "textShadowColor": "rgba(0, 0, 0, 0.4)",
              "textShadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "textShadowRadius": 2,
            },
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`<ToolbarButton /> renders correctly with onPress 1`] = `
<TouchableOpacity
  borderless={true}
  delayPressIn={0}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 6,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "padding": 6,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontSize": 24,
          },
          Array [
            undefined,
            false,
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
</TouchableOpacity>
`;
